codes/classical/matrices/rank_metric.yml

```diff
@@ -0,0 +1,38 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: rank_metric
+physical: matrices
+logical: q-ary_digits
+
+name: 'Rank metric code'
+introduced: '\cite{doi:10.1016/0097-3165(78)90015-8}'
+
+
+description: |
+  Encodes \(k\) states in an \(m\times \n)-dimensional \textit{matrix} over \(GF(q)\), using the rank of the difference of matrices as the distance function.
+  Distance \(d\) is the minimum rank matrix in the code.
+  Generally linear.
+
+
+protection: 'Protects against errors with rank \(\leq \lfloor \frac{d-1}2 \rfloor\).'
+
+
+notes:
+  - 'Number of codewords \(k \leq max(n, m) (min(n, m) - d + 1) \).'
+  - 'Codes that achieve the above bound with equality are Maximum Rank Distance (MRD).'
+  - 'Number of codewords \(k \leq max(n, m) M\), where \(M\) is the max rank matrix in the code.'
+  - 'Codes that achieve this second bound with equality are Delsarte optimal anticodes.'
```

Comment:
Codes that achieve this bound with equality are **called** Delsarte optimal anticodes.
(delete `second', insert `called')